admin/src/scenes/volontaires/components/phase1/DocumentPhase1.js

```diff
@@ -1,15 +1,29 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 import { toastr } from "react-redux-toastr";
-import FileIcon from "../../../assets/FileIcon";
-import api from "../../../services/api";
-import { SENDINBLUE_TEMPLATES, translate } from "../../../utils";
-import ButtonPlain from "./ButtonPlain";
-import { capture } from "../../../sentry";
-import Select from "../../../components/TailwindSelect";
+import FileIcon from "../../../../assets/FileIcon";
+import api from "../../../../services/api";
+import { SENDINBLUE_TEMPLATES, translate } from "../../../../utils";
+import { capture } from "../../../../sentry";
+import Select from "../../../../components/TailwindSelect";
+import ButtonPrimary from "../../../../components/ui/buttons/ButtonPrimary";
+
+const getStatusCohesionStayMedical = (young) => {
+  if (young) {
+    if (young.cohesionStayMedicalFileReceived !== "true") {
+      if (young.cohesionStayMedicalFileDownload === "false") {
+        return "TO_DOWNLOAD";
+      } else {
+        return "DOWNLOADED";
+      }
+    } else {
+      return "RECEIVED";
+    }
+  }
```

Comment:
Peut-être que ma suggestion n'est pas une bonne idée, mais j'ai eu du mal à lire ce passage : trois if imbriqués plus deux else (plus un cas non traité) pour 3 possibilités, ça faisait beaucoup niveau complexité cognitive.

```suggestion
  if (young?.cohesionStayMedicalFileDownload === "false") return "TO_DOWNLOAD";
  if (young?.cohesionStayMedicalFileDownload === "true") return "RECEIVED";
  return "DOWNLOADED";
```

Reply:
C'était déjà comme ça. Je n'ai fait que déplacer la logique qui était dans un `useEffect` pour éviter de la dupliquer. Mais oui je suis d'accord, c'est pas l'idéal